components/fxa-client/src/fxa_msg_types.proto

```diff
@@ -72,27 +72,50 @@ message Capabilities {
     repeated Device.Capability capability = 1;
 }
 
-// This is basically an enum with associated values,
-// but it's a bit harder to model in proto2.
-message AccountEvent {
-    enum AccountEventType {
-        TAB_RECEIVED = 1; // data set to TabReceivedData.
+message IncomingDeviceCommand {
+    enum IncomingDeviceCommandType {
+        SEND_TAB = 1; // `data` set to `send_tab_data`.
     }
-    required AccountEventType type = 1;
+    required IncomingDeviceCommandType type = 1;
 
-    message TabReceivedData {
+    message SendTabData {
         message TabHistoryEntry {
             required string title = 1;
             required string url = 2;
         }
         optional Device from = 1;
         repeated TabHistoryEntry entries = 2;
     }
+
     oneof data {
-        TabReceivedData tab_received_data = 2;
+        SendTabData send_tab_data = 2;
     };
 }
 
+message IncomingDeviceCommands {
+    repeated IncomingDeviceCommand commands = 1;
+}
+
+// This is basically an enum with associated values,
+// but it's a bit harder to model in proto2.
+message AccountEvent {
+    enum AccountEventType {
+        INCOMING_DEVICE_COMMAND = 1; // `data` set to `device_command`.
+        PROFILE_UPDATED = 2;
+        DEVICE_CONNECTED = 3; // `data` set to `device_connected_name`.
+        ACCOUNT_AUTH_STATE_CHANGED = 4;
+        DEVICE_DISCONNECTED = 5; // `data` set to `device_disconnected_is_local`.
+        ACCOUNT_DESTROYED = 6;
+    }
+    required AccountEventType type = 1;
+
+    oneof data {
+        IncomingDeviceCommand device_command = 2;
+        string device_connected_name = 3;
+        bool device_disconnected_is_local = 4;
```

Comment:
For my own understanding: does this schema allow a message with `INCOMING_DEVICE_COMMAND` as its type but a string device name as its `data` field? If we ever accidentally generated such a thing and sent it out over the FFI, what would happen on the receiving side?

Reply:
Yes it is a small risk, I think what we could do instead is use the `oneof data` as the enum itself, such as:
```
message AccountEvent {
    oneof data {
        IncomingDeviceCommand device_command = 1;
        string device_connected_name = 2;
        bool device_disconnected_is_local = 3;
        bool profile_updated = 4; // The boolean value is not used.
    }
}
```

We can experiment this technique in another patch.